Makefile

```diff
@@ -167,13 +190,25 @@ OBJS := \
 OBJS := $(addprefix $(OUT)/, $(OBJS))
 deps := $(OBJS:%.o=%.o.d)
 
+EXPORTED_FUNCS := _main
+EMCC_CFLAGS :=
+EMCC_MACRO :=
+ifeq ("$(CC_IS_EMCC)", "1")
+EMCC_MACRO += -DMEM_SIZE=0x40000000    # 1GB
+EMCC_CFLAGS += -sINITIAL_MEMORY=2GB \
+	       -sALLOW_MEMORY_GROWTH \
+	       --embed-file build \
+	       -s"EXPORTED_FUNCTIONS=$(EXPORTED_FUNCS)" \
+	       -sMALLOC=mimalloc
```

Comment:
Since you are using mimalloc, you have to check the versions of emcc which bundle the corresponding packages.